src/reference_implementations/prompting_vector_llms/llm_prompting_examples/llm_prompt_ag_news.ipynb

```diff
@@ -0,0 +1,1107 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 29,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "100%|██████████| 100/100 [10:50<00:00,  6.51s/it]\n"
+     ]
+    }
+   ],
+   "source": [
+    "prompt_template_postfix = \"Category (World, Sports, Business, Technology):\"\n",
+    "prompts = [\n",
+    "    f\"{prompt_demonstrations}{model_input_text}\\n{prompt_template_postfix}\" for model_input_text in model_input_texts\n",
+    "]\n",
+    "# For memory management, we split the prompts into batches of size 1, since the activations are heavier.\n",
+    "prompt_batches = split_prompts_into_batches(prompts, 1)\n",
+    "predicted_labels = []\n",
+    "for prompt_batch in tqdm(prompt_batches):\n",
+    "    activations = model.get_activations(prompt_batch, [last_layer_name], short_generation_config)\n",
+    "    for activations_single_prompt in activations.activations:\n",
+    "        last_layer_matrix = activations_single_prompt[last_layer_name]\n",
+    "        predicted_label = get_label_with_highest_likelihood(\n",
+    "            last_layer_matrix, label_token_ids, int_to_label_map, right_shift=True\n",
+    "        )\n",
+    "        predicted_labels.append(predicted_label)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Prediction Accuracy: 0.83\n",
+      "Confusion Matrix with ordering ['world', 'sports', 'business', 'technology']\n",
+      "[[17  1  7  3]\n",
+      " [ 0 18  3  0]\n",
+      " [ 1  0 21  1]\n",
+      " [ 1  0  0 27]]\n",
+      "========================================================\n",
+      "Label: world, F1: 0.7234042553191489, Precision: 0.8947368421052632, Recall: 0.6071428571428571\n",
+      "Label: sports, F1: 0.9, Precision: 0.9473684210526315, Recall: 0.8571428571428571\n",
+      "Label: business, F1: 0.7777777777777777, Precision: 0.6774193548387096, Recall: 0.9130434782608695\n",
+      "Label: technology, F1: 0.9152542372881356, Precision: 0.8709677419354839, Recall: 0.9642857142857143\n"
+     ]
+    }
+   ],
+   "source": [
+    "report_metrics(predicted_labels, ag_news_labels, labels_order=[\"world\", \"sports\", \"business\", \"technology\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Because we're using the likelihood mapping, there are no instances in which we fail to match our labels and we're performing this task quite well. Note that if we set our generation to greedy decoding (`temperature = 0`), we would likely get close to this performance without likelihood matching, but we'd still have label matching issues for some of our generations."
```

Comment:
Similarly, we can say a bit more why we expect greedy decoding to have this effect